### ISSUE TITLE
Any chance of making this work?

### ISSUE THREAD

**Issue by xiaodaigh** [CreRecombinase/libblosc#1]
BTW I love the API in https://github.com/stevengj/Blosc.jl

**Comment by CreRecombinase** [comment]
Kudos for finding this repo.  My main motivation for creating this package was as a way of vendoring c-blosc for a different package.  The problem I ran into is that building c-blosc relies on CMake.  This means that I can either (from least to most preferably):
1) Make c-blosc a [System Requirement](https://cran.r-project.org/doc/manuals/r-release/R-exts.html#Non_002dR-scripts-in-packages). Like I said above, the main goal of this package is that it provides an installation of blosc for other R packages, so that would kind of defeat (my) use case.
1) Call CMake during configure, and make CMake a [System Requirement]() . This also isn't great because most R users don't have CMake, so it doesn't really solve my original problem, just pushes it one layer down
2) Try to port the c-blosc build script to autotools.  I'm not an expert at either CMake or autotools, so this would be a slog.  The other unappealing part of this solution is that it would take continued effort to keep up with changes in the blosc build system (unless I could get it upstreamed, which I also kind of doubt). This is probably the most user friendly (if developer hostile) solution though, as it requires nothing on the user's part.

What I've done instead for my other project is to vendor zstd, (which has a much simpler build process and comparable performance) directly into my main project.  

What do you think?

**Comment by xiaodaigh** [comment]
OK. Yeah, like what's inside the fst package already?

**Comment by CreRecombinase** [comment]
Yeah exactly. .   For Makefile-based stuff adding C libraries to an R package is basically plug-and-play [I also played with the idea of making a dedicated package for that as well](https://github.com/CreRecombinase/rzstdlib).  Taking a look at the CMake in c-blosc again I really do think Option 3 is totally do-able, I just don't have the time right now.  Writing the actual R wrapper would actually be pretty straight-forward.

**Comment by CreRecombinase** [comment]
Good news, it looks like [there's a proof of concept for vendoring blosc in R:](https://github.com/grimbough/rhdf5filters) . In the coming weeks I'll take a crack at it, then I can see about writing the actual code to wrap the library

**Comment by xiaodaigh** [comment]
Thanks!!!
